source-map.bs

```diff
@@ -274,6 +290,44 @@ fields represent mapped code that also has a mapped name.
 Note: Using file offsets was considered but rejected in favor of using line/column data to avoid becoming
 misaligned with the original due to platform-specific line endings. 
 
+### [=names=] for generated JavaScript code
+
+Source map generators should create a [=mapping=] entry with a [=name=] field for
+the following JavaScript tokens, if
+
+1. The original source language construct maps semantically to the generated
+    JavaScript code.
+
+1. The original source language construct has a name.
+
+Then the [=name=] of the [=mapping=] entry should be the name of the original
+source language construct.
+
+Example: A TypeScript arrow function compiled to a JavaScript function, or a
+local variable that was renamed.
+
+1. The opening parenthesis `(` of [=FunctionDeclaration=], [=FunctionExpression=],
```

Comment:
Does this mean that the generator should add a mapping with a name pointing to `(`, even if the generated function also has a name? That would map the `name` to the token that follows the actual name. This seems to conflict with (5) where variables and parameter names are mapped to the first character of the identifier.

Reply:
Yes, for function/generator declarations with a `BindingIdentifier` in the generated code we basically recommend emitting a named mapping for both the `BindingIdentifier` as well as the `(`.

The advantage is that for consumers this is somewhat easier since they don't have to distinguish too many cases and encoding wise it's not too expensive as the name on the `(` will always be encoded as `A` (`0`) since it's the same name.

Reply:
I updated the text to read that generators "should" emit  the name on the mapping for `BindingIdentifier` or on the opening paren if no identifier exists. But they may emit it on the opening paren regardless.